cmds/record.c

```diff
@@ -134,6 +134,8 @@ char * get_libmcount_path(struct opts *opts)
 	snprintf(lib, PATH_MAX, "%s/%s", INSTALL_LIB_PATH, libmcount);
 	if (access(lib, F_OK) != 0 && errno == ENOENT)
 		pr_warn("Didn't you run 'make install' ?\n");
+	else
+		return lib;
```

Comment:
But in this case, it may return invalid lib path if the errno is not ENOENT?

Reply:
How about this?
```diff
diff --git a/cmds/record.c b/cmds/record.c
index 8b499b6..295399c 100644
--- a/cmds/record.c
+++ b/cmds/record.c
@@ -132,11 +132,14 @@ char * get_libmcount_path(struct opts *opts)
 
 #ifdef INSTALL_LIB_PATH
        snprintf(lib, PATH_MAX, "%s/%s", INSTALL_LIB_PATH, libmcount);
-       if (access(lib, F_OK) != 0 && errno == ENOENT)
-               pr_warn("Didn't you run 'make install' ?\n");
-#endif
+       if (access(lib, F_OK) == 0)
+               return lib;
+       free(lib);
+       return NULL;
+#else
        strcpy(lib, libmcount);
        return lib;
+#endif
 }
 
 void put_libmcount_path(char *libpath)
```

Reply:
How about this?

```diff
diff --git a/cmds/record.c b/cmds/record.c
index 8b499b6..3885759 100644
--- a/cmds/record.c
+++ b/cmds/record.c
@@ -132,8 +132,12 @@ char * get_libmcount_path(struct opts *opts)
 
 #ifdef INSTALL_LIB_PATH
        snprintf(lib, PATH_MAX, "%s/%s", INSTALL_LIB_PATH, libmcount);
-       if (access(lib, F_OK) != 0 && errno == ENOENT)
+       if (access(lib, F_OK) == 0)
+               return lib;
+
+        if (errno == ENOENT)
                pr_warn("Didn't you run 'make install' ?\n");
+        /* fall through */
 #endif
        strcpy(lib, libmcount);
        return lib;
```